src/smt/mod.rs

```diff
@@ -204,3 +299,36 @@ fn ty_to_sort<'ctx>(ctx: &SmtCtx<'ctx>, ty: &TyKind) -> Sort<'ctx> {
         }
     }
 }
+
+pub enum Lit<'ctx> {
```

Comment:
Maybe add a comment how this wraps z3rro's Lit type? And also explain what this is and why it's needed.